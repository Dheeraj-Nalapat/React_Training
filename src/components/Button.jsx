const Button = ({ text, type, id, handleSubmit }) => {
  return (
    <button type={type} id={id} onClick={handleSubmit}>
      {text}
    </button>
  );
};

export default Button;
