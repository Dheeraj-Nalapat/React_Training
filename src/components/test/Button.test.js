import { fireEvent, render } from "@testing-library/react";
import Button from "../Button";
import React from "react";

describe("Check if Button works properly", () => {
  test("first", () => {
    const { getByTestId } = render(<Button />);
    const element = getByTestId("button-test-id");
    console.log(element);
    expect(element).toBeTruthy();
  });
  test("check if text displayed properly", () => {
    const text = "click";
    const { getByText } = render(<Button text={text} />);
    getByText(text);
  });
  test("Check is onClick is triggered", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(<Button handleSubmit={onClick} />);
    const element = getByTestId("button-test-id");
    fireEvent.click(element);
    expect(onClick).toHaveBeenCalledTimes(1);
  });
  test("Check if snapshot is mathced properly", () => {
    const onClick = jest.fn();
    const text = "Click";
    const { asFragment } = render(
      <Button text={text} handleSubmit={onClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
