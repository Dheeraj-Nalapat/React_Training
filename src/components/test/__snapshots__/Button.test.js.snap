// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Button works properly Check if snapshot is mathced properly 1`] = `
<DocumentFragment>
  <button
    data-testid="button-test-id"
  >
    Click
  </button>
</DocumentFragment>
`;
